### PR TITLE
Fix Issue #41981"Make struct fields writable" needs to list the actual reason

### LINKED ISSUE #41981: "Make struct fields writable" needs to list the actual reason
I got the following with no indication as to "why". Needed to read the code of the thing to figure out why it was suggesting this: https://github.com/dotnet/roslyn/blob/c5f5bb705c5eafbc17126ccb6ab653bd0f15a71b/src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs#L53. The struct assigns "this" outside constructor.

image

### PR DESCRIPTION
**Pull request by Sliptory**
Fixes #41981

### CODE REVIEW

File: src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ internal sealed class CSharpMakeStructFieldsWritableDiagnosticAnalyzer : Abstrac
         private static readonly DiagnosticDescriptor s_diagnosticDescriptor = CreateDescriptor(
             IDEDiagnosticIds.MakeStructFieldsWritable,
             new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
-            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            new LocalizableResourceString(nameof(FeaturesResources.Struct_contains_assigment_to_this_outside_the_scope_constructor_Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs]: `Struct_contains_assignment_to_this_outside_of_constructor_Make_readonly_fields_writable`

File: src/Features/Core/Portable/FeaturesResources.resx [hunk 2]
```diff
@@ -1756,4 +1756,7 @@ This version used in: {2}</value>
   <data name="Implement_explicitly" xml:space="preserve">
     <value>Implement explicitly</value>
   </data>
+  <data name="Struct_contains_assigment_to_this_outside_the_scope_constructor_Make_readonly_fields_writable" xml:space="preserve">
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/FeaturesResources.resx]: `assigment` mispelled.

File: src/Features/Core/Portable/FeaturesResources.resx [hunk 3]
```diff
@@ -1756,4 +1756,7 @@ This version used in: {2}</value>
...
+    <value>Struct contains assignment to 'this' outside the scope of constructor. Make readonly fields writable</value>
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/FeaturesResources.resx]: remove `the scope`.

File: src/Features/Core/Portable/xlf/FeaturesResources.cs.xlf [hunk 4]
```diff
@@ -562,6 +562,11 @@
         <target state="translated">Stream musí podporovat operace read a seek.</target>
         <note />
       </trans-unit>
+      <trans-unit id="Struct_contains_assigment_to_this_outside_the_scope_constructor_Make_readonly_fields_writable">
+        <source>Struct contains assignment to 'this' outside the scope of constructor. Make readonly fields writable</source>
+        <target state="translated">Struktura obsahuje přiřazení k „tomuto“ mimo rozsah konstruktoru. Umožněte zapisování polí jen pro čtení</target>
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/xlf/FeaturesResources.cs.xlf]: did you manually translate these yourself? :)

you don't need to do that.  the loc team can do this here :)

Comment (Sliptory) [reply src/Features/Core/Portable/xlf/FeaturesResources.cs.xlf]: ok 😀

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/xlf/FeaturesResources.cs.xlf]: can you undo these :)  these should jus tbe generated as english.  loc will take over.

**Comment by CyrusNajmabadi** [comment]
@Sliptory are you ok with me making changes to your PR?

**Comment by Sliptory** [comment]
> @Sliptory are you ok with me making changes to your PR?

sure 👍

**Comment by CyrusNajmabadi** [comment]
Thanks for the contribution!

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**